homeassistant/components/dio_chacon/cover.py

```diff
@@ -0,0 +1,164 @@
+"""Cover Platform for Dio Chacon REV-SHUTTER devices."""
+
+import logging
+from typing import Any
+
+from dio_chacon_wifi_api import DIOChaconAPIClient
+from dio_chacon_wifi_api.const import DeviceTypeEnum, ShutterMoveEnum
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DioChaconData
+from .entity import DioChaconEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Defer sensor setup to the shared sensor module."""
+    dio_chacon_data: DioChaconData = config_entry.runtime_data
+
+    entities: list[DioChaconCover] = []
+    for device in dio_chacon_data.list_devices:
+        if device["type"] == DeviceTypeEnum.SHUTTER.value:
+            cover = DioChaconCover(dio_chacon_data.dio_chacon_client, device)
+            # Registers a callback to update the device state when the server sends an event
+            dio_chacon_data.dio_chacon_client.set_callback_device_state_by_device(
+                cover.target_id, cover.callback_device_state
+            )
+            entities.append(cover)
+
+    async_add_entities(entities)
+
+
+class DioChaconCover(DioChaconEntity, CoverEntity):
+    """Object for controlling a Dio Chacon cover."""
+
+    # To avoid having name twice
+    _attr_has_entity_name = False
```

Comment:
Right so the catch why you have the name twice is indeed because of `_attr_has_entity_name = True`. It will append the entity name to the device name, and since the name of this entity is the device name, it will append the device name to the device name. The solution to this:
```py
_attr_has_entity_name = True # but you already have this in entity.py, so don't place it here
_attr_name = None
```

Reply:
OK